perfkitbenchmarker/providers/gcp/gcs.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import re
+
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import linux_packages
+from perfkitbenchmarker import object_storage_service
+from perfkitbenchmarker import providers
+from perfkitbenchmarker import vm_util
+
+flags.DEFINE_string('google_cloud_sdk_version', None,
+                    'Use a particular version of the Google Cloud SDK, e.g.: '
+                    '103.0.0')
+
+FLAGS = flags.FLAGS
+
+GCS_CREDENTIAL_LOCATION = '.config/gcloud/credentials'
+DEFAULT_GCP_REGION = 'us-central1'
+
+
+class GoogleCloudStorageService(object_storage_service.ObjectStorageService):
+  """Interface to Google Cloud Storage."""
+
+  STORAGE_NAME = providers.GCP
+
+  def PrepareService(self, location):
+    self.location = location or DEFAULT_GCP_REGION
+
+  def MakeBucket(self, bucket):
+    command = ['gsutil', 'mb']
+    if self.location:
+      command.extend(['-l', self.location])
+    if self.location and '-' in self.location:
+      # regional buckets are always DRA
+      command.extend(['-c', 'DRA'])
+    elif FLAGS.object_storage_storage_class is not None:
+      command.extend(['-c', FLAGS.object_storage_storage_class])
+    command.extend(['gs://%s' % bucket])
+
+    vm_util.IssueCommand(command)
+
+  @vm_util.Retry()
+  def DeleteBucket(self, bucket):
+    # We want to retry rm and rb together because it's possible that
+    # we issue rm followed by rb, but then rb fails because the
+    # metadata store isn't consistent and the server that handles the
+    # rb thinks there are still objects in the bucket. It's also
+    # possible for rm to fail because the metadata store is
+    # inconsistent and rm doesn't find all objects, so can't delete
+    # them all.
+    self.EmptyBucket(bucket)
+
+    vm_util.IssueCommand(
+        ['gsutil', 'rb',
+         'gs://%s' % bucket])
+
+  def EmptyBucket(self, bucket):
+    vm_util.IssueCommand(
+        ['gsutil', '-m', 'rm', '-r',
+         'gs://%s/*' % bucket])
+
+  def PrepareVM(self, vm):
+    vm.Install('wget')
```

Comment:
I would like to, but I don't think our current `vm.Install` interface can handle installing specific package versions. I'd like to keep that change separate from this PR.